Navbar

alexodan opened the pull request:
## Learnings
<!-- Your major learnings and WOW moments during this challenge -->

- Probably the major one is the usage of `Context` to communicate between parent-child components that depend on each other, and you don't want to expose some functionality to the user. I tried to peek into the code of [Reach UI](https://github.com/reach/reach-ui) to get an idea, but it is very complex. I got just the basics and implemented it for `Navbar` and `NavbarItem`.

- There are CSS props that are **not** animatable, like `display`. I was, at first, trying to use it to animate the fading of the NavbarItem text and the selected icon, then switched it to use opacity.

- Testing with Cypress and CSS modules can get a bit tricky, the classes are dynamic so you can't do things like `cy.get('[data-testid="clock-icon"]').should("have.class", "active");`, because you get `_active_a123as` or something like that instead. You have to choose between dataset or custom attributes and get creative, https://github.com/cypress-io/cypress/issues/1212. It is possible also to strip the `data-test-id` attributes with a babel plugin https://github.com/coderas/babel-plugin-jsx-remove-data-test-id.

- A bunch of learning of accessibility, what a huge topic. Animations can be disabled from the OS, since people can feel dizzy by looking at them, among other things, so there is a media query that can be used to set the animations only if the user allow them `@media (prefers-reduced-motion: reduce)`, more info https://www.tatianamac.com/posts/prefers-reduced-motion.

- Naming conventions on CSS: https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md#is-stateOfComponent.

- Adding Percy as part of visual testing, which relies on snapshots captured and uploaded to the cloud.

## Struggles
<!-- What did you not understand, or struggled with the most -->

I'd say there were at least two things. I definitely struggled with the animation portion, I don't entirely understand how they work, the code is messy and the animation doesn't 'feel' entirely right.

Coming up with which things to expose in order to make a good component API, for example, the size of the icons, how to render them, or giving the option to change the colors. I also saw in Reach UI they use `forwardRef` quite extensively, and didn't understand what for, I believe it has to do with giving the user more fine-grained control of the component.

Another one is, I didn't find a way to completely hide the `id` prop from `NavbarItem` to the user of the component, maybe there is some other approach for this.

## References
<!-- Resources/links you found useful while doing the challenge -->

I didn't track all of them but here are a few I used:

- [Reach UI](https://github.com/reach/reach-ui)
- [Dark mode](https://www.joshwcomeau.com/react/dark-mode/)
- [CSS fade in animation](https://blog.hubspot.com/website/css-fade-in)

alexodan committed:
Adds navbar component. Adds scss variables for styling. Adds navbar example on App component.

alexodan committed:
Adds theme context.

alexodan committed:
Improving accessibility

alexodan committed:
Merge with main

alexodan committed:
Adds clone element to get the id of each item.

I'm regretting using typescript i dunno why i chose it halp pls

alexodan committed:
Add some logic to track the position of the selected item indicator icon

- Add some logic to track the position of the selected item indicator icon
- Improves some variable names

alexodan committed:
Start working on the animation

alexodan committed:
Added a very hacky way to do the dot transition

alexodan committed:
Cleaning anys used in refs

alexodan committed:
Tiding up some code

alexodan committed:
Adds animation for text to disappear when item is selected

alexodan committed:
Changes font. Removes  unused code to toggle dark/light mode.

sandrina-p commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,183 @@
+              });
+            })}
+            {selectedId !== undefined && (
+              <div
```
bug: [ul > div is invalid html](https://caninclude.glitch.me/caninclude?child=div&parent=ul). this dot needs to be a child of nav.

alexodan replied:
oh nice catch,
btw i love this site (didn't know it) and specifically this thing
<img width="134" alt="image" src="https://github.com/alexodan/navbar-challenge/assets/35075089/eff113fa-a7f1-4fc4-a8d4-2ec44ec29425">

sandrina-p commented on src/utils.ts:
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+export function isNavbarItem(child: React.ReactNode): boolean {
```
In real projects, when a component is built, every file related to it is under the same folder, here's an example:

```
components/
  Navbar
    Navbar.tsx
    Navbar.scss
    Navbar.tests..tsx
    Navbar.utils.ts
    Navbar.*.ts 
```

This way it's much easier to maintain the component and move it no other folders/projects if needed. Gotcha?

sandrina-p commented on src/theme.tsx:
```diff
@@ -0,0 +1,37 @@
+function getInitialColorMode() {
+  const mql = window.matchMedia("(prefers-color-scheme: dark)");
+  const hasMediaQueryPreference = typeof mql.matches === "boolean";
+  if (hasMediaQueryPreference) {
```
praise: Well done with the theming setup here!

bug: It's just missing the mq listener, in case the user changes the theme while the page is running.

sandrina-p commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,183 @@
+    const domMenuItem = items[selectedId!]?.current;
+    const domSelectedIndicator = selectedIndicatorRef.current;
+    if (domMenuItem && domSelectedIndicator) {
+      animateWithCss(domMenuItem, domSelectedIndicator);
```
bug: It works well in the initial render, but if the component re-renders with just 3 or 5 items, it breaks. It's missing a dependency in one of the hooks ;)

alexodan replied:
fixed :raised_hands:

sandrina-p commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,183 @@
+  const [items, setItems] = useState<React.RefObject<HTMLLIElement>[]>([]);
+  const selectedIndicatorRef = useRef<HTMLDivElement>(null);
+
+  // TODO: refactor this monstrosity (maybe ask Sandrina to give some tips)
```
_make it work, then make it better_.  Indeed it can be reduced, quite a lot. First, I'd recommend you to:
- extract this logic to its own hook eg `useDotAnimation`
- Avoid accessing the DOM directly. Instead store the styles in useState and pass them down as: `<div ref={selectedIndicatorRef} style={dotStyles}> ...</div>` 

I'll review this part again later tonight/tomorrow, as I have to jump in a call now

sandrina-p commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,183 @@
+
+  return (
+    <li
+      tabIndex={0}
```
question: Tell me, why do we need this `tabIndex`? What's the downside?

Hint: It would be better using `li > button`, or, `li > a` as these are accessible by default.

alexodan replied:
changed it to use button for accessibility :accessibility:

sandrina-p replied:
Great! **Now the next challenge:** What if a dev uses the component where one of the items goes to another page. How do you adapt the component to support both cases (button or link)?

alexodan replied:
I went for using the `as` prop to pass `a` or `button` (with button as default)

sandrina-p commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,183 @@
+                throw Error("Only NavbarItem allowed as child of Navbar");
+              }
+              return React.cloneElement(child as React.ReactElement, {
+                id: index,
```
bug: You might already know that we should avoid [`index` for unique ids](https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66) (or even `key` prop in arrays) has its downsides when the component re-renders. The same applies here. Ideally, the good practice here is to force the user to pass a `id` required prop for each NavItem.

alexodan replied:
hehe yes, at least in my mind, this navbar had a fixed number of items, so I didn't consider this a problem. Also now that you mention it, I spent some time thinking about if it made sense to expose something like an id and didn't arrive at a strong conclusion

sandrina-p replied:
Usually, yes a navbar has a fixed number, but think about SPAs, it's not that uncommon for the navbar to change based on user permissions, login/logout, etc etc ;)

sandrina-p replied:
Here's another advantage of passing an id:  How can the user (dev) set a default active item?

sandrina-p replied:
Have you tacked this one about the id?

sandrina-p replied:
Ah, nevermind, i noticed now the  `defaultActive={0}`.

sandrina-p commented on src/components/Navbar.tsx:
```diff
@@ -0,0 +1,183 @@
+        <nav aria-label={props.label}>
+          <ul role="menubar" className="navbar">
+            {React.Children.map(props.children, (child, index) => {
+              if (isNavbarItem(child)) {
```
praise: Well done with this check! 

One extra tip: In real projects, it's common to use env vars to conditionally throw these errors. For example, if you notice, React has a bunch of dev warnings that don't exist in production. This way the app doesn't explode in prod, and the bundle size is smaller (thanks to three-shaking)

You can check [vite's env vars](https://vitejs.dev/guide/env-and-mode.html), and apply it like this:


```
// utils.js
const isDev = () => import.meta.env.MODE === "development"

if (isDev() && isNavbarItem(child)) { ... }
```

alexodan replied:
added, also fixed the check that had a bug on a closer look 😵‍💫

alexodan committed:
Adds buttons to add/remove mock item (to test animation issue easily)

alexodan committed:
Moving Navbar component to separate folder and Fixes animation

- Fixes animation and select/deselect when an item is removed then added.
- Introduces Navbar folder and moves components and relative files to the same path.
- Improves (fixes) the check isNavbarItem()
- Adds flag to only throw error in development mode.

alexodan committed:
Refactoring animation to hook.

alexodan committed:
Move defaultSelected to Navbar. Fixes off by one animation error.

alexodan committed:
Refactor animation with CSS. Renamed selected for active, sounds better.

alexodan committed:
Added storybook

alexodan committed:
Updates animation to fill icons

alexodan committed:
Adds style override prop for icons. Change logic of selecting already selected item.

alexodan committed:
Adds cypress initial config and tests

alexodan committed:
Cypress command and ignore cypress videos

alexodan committed:
Adding cypress github action

alexodan committed:
Update readme to test gh actions

alexodan committed:
Fixes cypress command. I think.

sandrina-p commented on src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,177 @@
+  icon: IconDefinition;
+  title: string;
+  onSelect: ({ title, id }: { title: string; id?: number }) => void;
+  id?: number; // TODO: find out how not to expose this prop
```
question: Why do you need to expose this prop?

alexodan replied:
I'm making TS happy 😅

sandrina-p replied:
Ah nevermind, I misunderstood this line 🙈

sandrina-p commented on src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,177 @@
+  return (
+    <li className="navbar-item" ref={itemRef}>
+      <button
+        onKeyDown={handleKeyDown}
```
question/todo: Are you sure you need this?

alexodan replied:
what?!?! 🤯 I feel Jon Snow, I really know nothing (removed since is not needed)
edit: that's because I'm using buttons now right?

sandrina-p replied:
Exactly! see, so many benefits from writing semantic HTML ^.^

sandrina-p commented on src/components/Navbar/navbar.scss:
```diff
@@ -0,0 +1,121 @@
+.navbar {
+  max-width: 480px;
+  position: relative;
+  .space {
```
todo: In CSS, avoid nested classes whenever possible, as it makes hard to do any needed future override. Instead prefer to concat classes names. (eg ass you do below with .navbar-item-btn)

Even better, this should be a `*.modules.scss` file to use CSS Modules and scope these classes, which allows you to keep the class name short without nesting.

Does this make sense to you?

alexodan replied:
until now I thought by using scss you didn't have to worry about scope 😵‍💫 I guess I haven't used css alone as much as styled components or tailwind
but yes, it makes sense!

sandrina-p replied:
exactly, SCSS does not solve scope. That's solved by CSS Modules (or css-in-js)

sandrina-p commented on src/components/Navbar/hooks.ts:
```diff
@@ -0,0 +1,18 @@
+import { useRef } from "react";
```
todo: The name should be "Navbar.hooks.ts"

alexodan replied:
done! :ok_hand:

sandrina-p commented on src/components/Navbar/hooks.ts:
```diff
@@ -0,0 +1,18 @@
+  const spaceRef = useRef<HTMLDivElement>(null);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  listRef.current?.style.setProperty("--navbar-items", "" + items);
```
bug/todo: Let's avoid using the direct DOM ref to set the class. Instead, it should be applied in the virtual DOM, as you had before. 

This hook only needs to return `{ style }` with all the variables, and then you pass them the `<ul style={style} />` the vars there automatically work in any child element.

You don't even need useRef here. Gotcha?

alexodan replied:
Gotcha! funny thing I had to cast the object with `as React.CSSProperties` otherwise TS would keep complaining.
This refactor is to reduce complexity, right? It has no effect on performance? 🤔

sandrina-p replied:
Exactly, don't expect any downside with performance. And yes, `React.CSSProperties` is great to ensure you pass valid CSS properties!

TS Question: Why do you need `String()` in `"--navbar-items": String(items)`? 🤔

alexodan replied:
I don't haha. Updated (maybe it was before doing the as React.CSSProperties)

sandrina-p commented on src/components/Navbar/navbar.scss:
```diff
@@ -0,0 +1,121 @@
+  &-icon {
+    transform-origin: bottom left;
+    &.active {
+      animation: jump 0.5s;
```
praise: nice touch!!

challenge: Here's one thing about [A11Y and Motion](https://www.tatianamac.com/posts/prefers-reduced-motion). Can you take it?

alexodan replied:
nice to know that, added!

sandrina-p replied:
it's perfect, well done! 

One extra tip for the future: `reduced-motion` != `no-motion`. You may still have animations on a website with reduced motion, just be mindful for those to be minimal/soft/subtle.

sandrina-p commented on src/components/Navbar/navbar.scss:
```diff
@@ -0,0 +1,121 @@
+.navbar-item-title {
+  display: block;
+  margin-top: 10px;
+  &.active {
```
tip: When using "state" classes that can be togglable, I like to call them "is*" to easily know that it's a conditional state. eg `isActive` this is inspired by the [SUIT CSS metholodogy](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md#is-stateOfComponent)

alexodan replied:
makes sense! updated it

sandrina-p commented on src/stories/Navbar.stories.tsx:
```diff
@@ -0,0 +1,65 @@
+import type { Meta, StoryObj } from "@storybook/react";
```
todo: Similar to all the other files, a component story would also be at `components/**/**.storires.tsx`. That way it's super clear to find everything about the component.

alexodan replied:
gotcha!

sandrina-p commented on src/stories/Introduction.mdx:
```diff
@@ -0,0 +1,205 @@
+import { Meta } from "@storybook/blocks";
```
todo: I believe you can delete this page, right? :)

sandrina-p replied:
@alexodan I think you forgot these 3 comments (github collapsed them in the feed 🙃 )

alexodan replied:
yep I did hehe, deleted it

sandrina-p commented on src/components/Navbar/Navbar.cy.tsx:
```diff
@@ -0,0 +1,89 @@
+];
+
+describe("<Navbar />", () => {
+  it("renders navbar with 4 items", () => {
```
todo: This test you say that it renders with 4 items, but it's missing an assertion to actually test 4 items in the DOM ;)

alexodan replied:
hehe, updated here 0e579e7

sandrina-p commented on src/components/Navbar/Navbar.cy.tsx:
```diff
@@ -0,0 +1,89 @@
+    );
+  });
+
+  it("should set first item as default active", () => {
```
todo: In this test you assert the `defaultActive`, but in the previous test you didn't test that there's no active item. When you have these "conditional" states, make sure to test both scenarios. In other words, the first test should say "renders navbar with 4 items, all non active"

alexodan replied:
nice!

sandrina-p commented on src/components/Navbar/Navbar.cy.tsx:
```diff
@@ -0,0 +1,89 @@
+
+    cy.get("@onClickSpy").should("have.been.called");
+    cy.get(".spot .dot").should("exist");
+    cy.get('[data-testid="clock-icon"]').should("have.class", "active");
```
suggestion: to go a step even further, you could test that when clicking another item (eg the 3rd one), the dot moves to the expected position. (assert the pixels here).

P.S. This test is considered a "visual test" which, normally is done with tools like [Percy](https://percy.io/), if you wanna give it a shot. (you can integrate it with Cypress!)

alexodan replied:
added Percy 💪, it was weird at first since I was expecting to assert things using code that looks more like Cypress, but Percy just relays on the snapshots u take (at least from what I understood) and compares them with the images that are uploaded

sandrina-p replied:
Yeah, exactly. Percy is just visual comparison, nothing more. Maybe with ChatGPT it will be more powerful to describe the differences eheh

sandrina-p commented on src/theme.tsx:
```diff
@@ -0,0 +1,65 @@
+        rawSetColorMode("light");
+      }
+    };
+    mql.addEventListener("change", toggle);
```
praise: Nice to see you trying to implement your custom mq listener!  In case you don't know it, check out [usehooks-ts](https://usehooks-ts.com/react-hook/use-media-query), which includes a lot of popular hooks done properly ;)

sandrina-p commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,26 @@
+name: Cypress Tests
```
note: This [crash course](https://youtu.be/1oJQRlz1v94) helped me understand better Github actions. Jump to min ~12 to explain the YML syntax.

sandrina-p reviewed (commented):
This is pretty much ready to go! I'd say there's only 2 missing things:
- a README about the repo, with a screenshot, features, etc
- publish the app! (or the storybook if you prefer)

alexodan committed:
Updates tests to assert default active and elements rendered. Updated hook to return style instead of using useRef. Added scss as module and refectored code.

alexodan committed:
Fixing component tests to accommodate scss modules

alexodan committed:
Adds storybook deploy scripts. Adds a11y first approach to animations.

alexodan committed:
Amplifies example code using Navbar

alexodan committed:
Adds button or link as prop. Adds percy.

alexodan committed:
Fixes as prop default value. Refactors scss class names to camelcase. Percy snapshot tests.

alexodan committed:
package-lock. should fix pipeline

sandrina-p commented on src/variables.scss:
```diff
@@ -1,7 +1,12 @@
+// Body
```
question: I forgot to ask this in the last review, but... why SCSS variables? You know native CSS variables right? (eg `--light-text: #000`).

Nowdays, with those, there's no point in using SCSS variables. You just define them at `app.scss` and then it's available globally in all files. No need for `variables.module.scss` either.

alexodan replied:
no reason in particular, I just thought I was using SCSS so => let's have SCSS variables 😅 it makes sense to move them to app.scss though given the user would set them, I would just provide some defaults somehow 🤔

alexodan replied:
i think it would be something like this? --primary would be defined by the user and #4c21ea would be the fallback I define

variables.scss
```
$primary: var(--primary, #4c21ea);
```

sandrina-p replied:
Exactly, yes, that's how you declare fallbacks!

P.S. SCSS in my opinion wins because of nesting. But even that is coming soon [Nesting in CSS](https://youtu.be/buChHSdsF9A?t=550)!

sandrina-p commented on src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,177 @@
+}>({});
+
+type NavbarItemProps<T extends string> = {
+  as: "a" | "button";
```
bug: This is fragile. Imagine the dev only passes the `a` without `href`. The DOM is invalid and the link won't work, but the app won't throw any error either. In other words, both `a` and `href` are required together, otherwise, the app should break somehow. How would you solve that?

alexodan replied:
added discriminated unions to handle it at type level

sandrina-p replied:
It's better but have you really tested? Because I can still pass invalid states. For example:
- If I pass `as="a"` it doesn't complain that the "href" is missing.
- If I pass `as="button" href="foo"` it doesn't complain that "href" is not allowed.

How would you solve that?

Give it a try first. Then, you can expand the solution:

<details>

```ts
// Bonus: Reuse the common props to avoid WET
type NavbarItemCommonProps<T extends string> = {
  icon: IconDefinition;
  title: T;
  onSelect: ({ title, id }: { title: T; id?: number }) => void;
  id?: number;
  iconStyles?: React.CSSProperties;
};

type NavbarItemProps<T extends string> =
  | ({
      as: "a";
      href: string;
    } & NavbarItemCommonProps<T>)
  | ({
      as: "button";
      href?: never; // "Discriminated union" - disallow a given param.
    } & NavbarItemCommonProps<T>);
```

</details>

alexodan replied:
<img width="741" alt="image" src="https://github.com/alexodan/navbar-challenge/assets/35075089/425dc923-9182-45a4-9332-87c7ed75534c">
<img width="879" alt="image" src="https://github.com/alexodan/navbar-challenge/assets/35075089/2910539a-510d-42e8-adca-6ec65644141c">
the errors themselves are super confusing but I get them when I try to pass button and href or "a" without href 🤔 I don't think is ideal tho!

alexodan replied:
I think I might be getting the error but for a different reason, since href should be never if as is button type

sandrina-p replied:
I 100% agree that most TS errors are confusing 😅 The alternative is to do the `isDev()` trick throwing a custom error message, without TS.

sandrina-p commented on src/App.tsx:
```diff
@@ -1,9 +1,94 @@
+    setActiveTab(title);
+  };
+
+  const getContentTab = (activeTab: Tab) => {
```
suggestion: We talked about this in a call last week. How can you improve this with O1 instead of On? (a direct lookup, remember?)

Also, as this variable does not depend on anything from React, it could be declared outside the component so it doesn't get redefined on every re-render.

alexodan replied:
hehe i didn't pay much attention to it since it was an example but fair enough, even examples are supposed to be good quality! 😆

sandrina-p replied:
In general, examples/docs must be the best quality, you know why? It's where most people copy from to their own code! :p  You don't want copilot to learn from your poor-quality examples ahah

sandrina-p commented on src/components/Navbar/Navbar.cy.tsx:
```diff
@@ -0,0 +1,107 @@
+      .should("contain", "active");
+  });
+
+  it("should fire an event and change active items on consecutive clicks", () => {
```
nice tests! 🙌

sandrina-p commented on package.json:
```diff
@@ -7,23 +8,53 @@
+    "build-storybook": "storybook build",
+    "cypress:open": "cypress open",
+    "cypress:run:component": "cypress run --component",
+    "percy": "percy exec -- cypress run --component --config-file cypress.config.ts"
```
bug/suggestion: I can't run percy locally because it's missing a token. I recommend you to add a guide about testing/percy on the README (and if possible share a token with me on discord just for testing purposes). The bonus would be to connect percy to CI ;)

alexodan replied:
added percy to CI 😎

sandrina-p commented on src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,177 @@
+            color={
+              isActive
+                ? iconStyles?.color ?? style.primary
+                : theme?.colorMode === "dark"
```
todo: You don't need JS for this theme conditional. Check the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) for CSS ;)

alexodan replied:
nice 😎 updated

sandrina-p replied:
sweeet ✨

sandrina-p commented on src/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,177 @@
+  useEffect(() => {
+    registerItem?.(itemRef);
+    return () => unRegisterItem?.(itemRef);
+  }, [setActiveId, id, registerItem, unRegisterItem]);
```
question: Why do we need setActiveId and id here as deps?

alexodan replied:
they were extra 😅 removed

alexodan committed:
Scss variables turned into css. Added type check for href prop. Refactored example tabs code. Added percy to CI maybe. Extra dependencies in use effect removed.

alexodan committed:
Adds percy token to github actions yml

alexodan committed:
Installing deps as part of CI

alexodan committed:
devops is hard yo

alexodan committed:
skipping install in gh actions

sandrina-p commented on readme.md:
```diff
@@ -7,3 +7,7 @@ This repo contains the solution to this challenge https://www.notion.so/alexdan/
 The general idea is to have a navbar like the one in this image
 
 ![Navbar dribbble gif](https://cdn.dribbble.com/users/414694/screenshots/5487895/media/61c4a366fd91e33a29bdca6cfdf5f68d.gif)
+
```
last todo: Let's rewrite this README to be a nice project showcase for your CV. Suggestion:

- about: short description + demo link + screenshots (eg gif showing the animation (i use [giphycapture](https://giphy.com/apps/giphycapture))
- features: (the things you've done: reusable, dark theme, motion, storybook, cypress percy, etc). and tooling (react, ts, context, etc)
- Usage: A small code snippet of its used.
- local setup: the steps on how to install, run, and test. (don't forget to document the Percy key!)

alexodan replied:
the percy key should be an example key or can I put my own?

alexodan replied:
nevermind 😆 `To enable Percy, the environment variable, PERCY_TOKEN, must be configured in your CI service. This is our write-only API token unique for each Percy project and should be kept secret.`
what do u mean with documenting the percy key, like how to create one or something?

sandrina-p replied:
Ah, I see, if it's private, then pls don't paste it here. By documenting, a link pointing to the guide explaining how to create/use it is enough

alexodan committed:
Renamed class based on SUIT. Removed introduction story. Moved navbar story.

alexodan committed:
Halfway through updating README. Added navbar width css variable.

alexodan committed:
Updates tests from active to isActive css classname

alexodan committed:
Default variables set in navbar. Updated type navbaritem.

sandrina-p commented on readme.md:
```diff
@@ -1,13 +1,85 @@
+- 🌑 / ☀️ Dark and Light theme
+- 🧱 Reusable
+- 📷 Automated visual tests with Cypress & Percy
+- ♿️ Motion a11y approach
```
nit: this is not a "a11y approach", it's a feature for a11y. like this: "Reduced motion for A11Y" or simply "Reduced motion" (with a link pointing to an article talking about it)

sandrina-p commented on readme.md:
```diff
@@ -1,13 +1,85 @@
+
+## Usage:
+
+A small code snippet of its used.
```
grammar: usage, not used. :)

sandrina-p commented on readme.md:
```diff
@@ -1,13 +1,85 @@
+function App() {
+  const [activeTab, setActiveTab] = useState<Tab>(initialItems[0].title);
+
+  const handleItemClicked = ({ title, id }: { title: Tab; id?: number }) => {
```
here id is not optional, right? I believe it must be required, otherwise, the dev will have trouble identifying the item.

sandrina-p commented on readme.md:
```diff
@@ -1,13 +1,85 @@
+          <NavbarItem
+            key={`${item.title}-${i}`}
+            icon={item.icon}
+            iconStyles={{ fontSize: "30px" }}
```
question: is this `iconStyles` required/essential for the smallest usage example?

alexodan replied:
Great question, and no, this is because the icons look very tiny if I don't add that, instead I put iconStyles as defaultProps. Not sure if it's the right call.

sandrina-p replied:
I'd say using default props is better as it's one less thing for the user/dev to worry about.

alexodan committed:
Updates README. Adds iconStyles as defaultProps. Makes id prop mandatory onSelect icon.

alexodan committed:
Updates readme command. Fix onClick.

alexodan committed:
Updated readme with docs and demo site link

sandrina-p commented on readme.md:
```diff
@@ -2,7 +2,7 @@
 ## Site
 
-Hosted using github pages [here](https://pages.github.com/)
+A demo is hosted using github pages [here](https://alexodan.github.io/navbar-challenge/)
```
tip:  [avoid click here links](https://uxmovement.com/content/why-your-links-should-never-say-click-here/)

```suggestion
Checkout the [demo page](https://alexodan.github.io/navbar-challenge/).
```

saying where it's hosted it's not relevant. Seeing a shiny link saying "demo" is the juicy part

alexodan committed:
Update readme.md

Co-authored-by: Sandrina Pereira <sandrin4pereir4@gmail.com>

sandrina-p reviewed (approved):
🎉 🎉 🎉
